kapitan/targets.py

```diff
@@ -349,6 +349,25 @@ def valid_target_obj(target_obj):
                     },
                     "required": ["input_type", "input_paths", "output_path"],
                     "minItems": 1,
+                    "oneOf": [
+                        {
+                            "properties": {
+                                "input_type": {
+                                    "enum": ["jsonnet", "kadet"] 
+                                },
+                                "output_type" : {
+                                    "enum": ["yaml", "json"]
+                                }
+                            },
+                        },
+                        {
+                            "properties": {
+                                "input_type": {
+                                    "enum": ["jinja2"]
+                                } 
+                            }
+                        }
+                    ],
```

Comment:
thanks @pranavgupta1234 

I find it a bit more legible if we add the `enum` fields inside `compile.items.properties`. what do you think ?

Something like this:

```
                "items": {
                    "type": "object",
                    "properties": {
                        "input_paths": {"type": "array"},
                        "input_type": {
                            "type": "string",
                            "enum": ["jsonnet", "kadet", "jinja2"]
                        },
                        "output_path": {"type": "string"},
                        "output_type": {
                            "type": "string",
                            "enum": ["yaml", "json"]
                        },
                    },
                    "required": ["input_type", "input_paths", "output_path"],
                    "minItems": 1,
```

Reply:
This is much cleaner and I also think the conditional logic is not required right now. Just a small reminder that the above jsonschema will not pass any arbitrary output file format with jinja2, i.e if I supply output format as xml with jinja2 it will fail while we can ignore this output format and let it pass which also happens currently and this property will break. Let me know your thoughts on this.

Reply:
The jsonschema I have originally proposed ignores any output_type for jinja2 and accepts json and yaml (will add plain too) for jsonnet and kadet.

Reply:
any updates @gburiola ?